### DISCUSSION TITLE
how to set emailVisibiltiy to true?

### DISCUSSION THREAD

**Discussion by Qa5imm** [pocketbase/pocketbase#3668] [Q&A] [answered]
hello,
I have two collections `registrations`, `users` and there's a direct relation between two. I am fetching user associated with a particular relation as below:
courseRegistrations = await pb.collection('registrations').getFullList({
            filter: `course_id= "${params?.id ?? ''}"`,
            expand: 'user_id',
        })
 everything's working fine except that I also need user email in the response. I have to make `emailVisibility = true` but I don't know how to do it. I have checked the user's collection and there's no emailVisibility field there that I can set to true, I tried setting the search rule to `emailVisibility=true` but that didn't work either. I would really appreciate if someone could help.
 here's my schema
 
[pb_schema (1).json](https://github.com/pocketbase/pocketbase/files/13250713/pb_schema.1.json)

**Comment by ganigeorgiev** [comment, marked as answer]
`emailVisibility` is a system field that comes with the auth collection and every auth record can manage their own email visibility.

When using the web APIs the field can be set during the create/update request:
```js
await pb.collection("users").update("RECORD_ID", {
    emailVisibility: true,
})
```

From the Admin UI you can manage it via the "Public: On/Off" email field toggle.
![email_field_screenshot](https://i.imgur.com/zZbF5oj.png)

**Reply by YugoCode** [reply]
@ganigeorgiev But wouldn't it be nice if the admin can toggle the emailVisibilty field through the UI (PB dashboard) for every user?

**Reply by ganigeorgiev** [reply]
Why would you want to do that? I'm not sure what would be the use case for this.
The idea of the `emailVisiblity` field is to allow users to control on their own whether they want their email to be publicly visible or not. You can always make it the default choice when creating the account or if you already have inserted user records and want to overwrite their field values you can create a migration that will execute SQL like `UPDATE users set emailVisibility=true`.

**Reply by YugoCode** [reply]
I agree with you, I just had a bunch of users with emailVisibilty not set to true but requiring it for the app. But using the migration SQL query and enabling it by default during account creation is the way to go, thanks! :)

**Reply by Qa5imm** [reply]
it worked, thanks
